libs/community/langchain_community/vectorstores/pgvector.py

```diff
@@ -508,7 +631,134 @@ def _results_to_docs_and_scores(self, results: Any) -> List[Tuple[Document, floa
+                f"{field}"
+            )
+
+        # Allow [a-zA-Z0-9_], disallow $ for now until we support escape characters
+        if not field.isidentifier():
+            raise ValueError(
+                f"Invalid field name: {field}. Expected a valid identifier."
+            )
+
+        if isinstance(value, dict):
+            # This is a filter specification
+            if len(value) != 1:
+                raise ValueError(
+                    "Invalid filter condition. Expected a value which "
+                    "is a dictionary with a single key that corresponds to an operator "
+                    f"but got a dictionary with {len(value)} keys. The first few "
+                    f"keys are: {list(value.keys())[:3]}"
+                )
+            operator, filter_value = list(value.items())[0]
+            # Verify that that operator is an operator
+            if operator not in SUPPORTED_OPERATORS:
+                raise ValueError(
+                    f"Invalid operator: {operator}. "
+                    f"Expected one of {SUPPORTED_OPERATORS}"
+                )
+        else:  # Then we assume an equality operator
+            operator = "$eq"
+            filter_value = value
+
+        # Now that we know the operator and the value we can create the filter
+        json_path_operation = self.EmbeddingStore.cmetadata.op("@@")
+
+        if operator in COMPARISONS_TO_NATIVE:
+            # Then we implement an equality filter
+            # native is trusted input
+            native = COMPARISONS_TO_NATIVE[operator]
+            if isinstance(filter_value, str):
+                # Note: dangerous
+                # We're sanitizing the input here ourselves because it seems
+                # impossible to get SQLAlchemy to double quote the string
+                # rather than single quote.
+                # we need the double quote for the jsonpath operations.
+                sanitized_value = _sanitized_double_quoted_string(filter_value)
+                path_value = sqlalchemy.text(
+                    f"'$.{field} {native} {sanitized_value}'",
+                )
+            else:
+                placeholder_name = f"filter_value_{counter.increment()}"
+                path_value = sqlalchemy.text(
+                    f"'$.{field} {native} :{placeholder_name}'",
+                ).bindparams(**{placeholder_name: filter_value})
+            return json_path_operation(path_value)
+        elif operator in "$between":
+            # Use AND with two comparisons
+            low, high = filter_value
+            placeholder_name = f"value_{counter.increment()}"
+            lower_bound = sqlalchemy.text(
+                f"'$.{field} >= :{placeholder_name}'",
+            ).bindparams(**{placeholder_name: low})
+
+            placeholder_name = f"value_{counter.increment()}"
+            upper_bound = sqlalchemy.text(
+                f"'$.{field} <= :{placeholder_name}'",
```

Comment:
is it standard for between to mean inclusive on both sides

Reply:
I think that's the convention in SQL